reward/staking_manager.go

```diff
@@ -47,17 +52,34 @@ type StakingManager struct {
 	blockchain           blockChain
 	chainHeadChan        chan blockchain.ChainHeadEvent
 	chainHeadSub         event.Subscription
+	migrationCheck       migrationCheck
 }
 
-func NewStakingManager(bc blockChain, gh governanceHelper, db stakingInfoDB) *StakingManager {
-	return &StakingManager{
+func NewStakingManager(bc blockChain, gh governanceHelper, db database.DBManager) *StakingManager {
+	sm := &StakingManager{
 		addressBookConnector: newAddressBookConnector(bc, gh),
 		stakingInfoCache:     newStakingInfoCache(),
-		stakingInfoDB:        db,
+		stakingInfoDB:        stakingInfoDB(db),
 		governanceHelper:     gh,
 		blockchain:           bc,
 		chainHeadChan:        make(chan blockchain.ChainHeadEvent, chainHeadChanSize),
+		migrationCheck:       migrationCheck(db),
 	}
+
+	// Before migration, staking information of current and before should be stored in DB.
+	//
+	// Staking information from block of StakingUpdateInterval ahead is needed to create a block.
+	// If there is no staking info in either cache, db or state trie, the node cannot make a block.
+	// The information in state trie is deleted after state trie migration.
+	blockchain.RegisterMigrationPrerequisites(func(blockNum uint64) error {
+		if _, err := sm.updateStakingInfo(blockNum); err != nil {
```

Comment:
If `blockNum` is not a `stakingInterval`. There is not thing happen.

Please consider below code.
```
sm.GetStakingInfo(ev.Block.NumberU64() + params.StakingUpdateInterval())
sm.GetStakingInfo(ev.Block.NumberU64())
```

Reply:
Thank you :) I changed it

Reply:
It think `GetStakingInfo` is propser for this. `updateStakingInfo` can works only `stakingInterval`.

Reply:
`GetStakingInfo` checks if staking info is stored either in cache, DB or address book. I want to make sure if it is stored in DB.
If it is `GetStakingInfo`, it will a problem for this cases:
- staking info is stored in cache
- staking info is not stored in cache and DB, but fails to store it in DB when `updateStakingInfo` is called.

How about I put `stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)` in `updateStakingInfo()`?

Reply:
As I mentioned https://github.com/klaytn/klaytn/pull/518/files#r428992343.
This way makes difference of the `GetStakingInfo` result.

How about make a new function for this?

Reply:
Thank you for the review. It is new applied with a new function :)